pkg/controller/projects/projects/project.go

```diff
@@ -149,6 +158,52 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errNotProject)
 	}
+	// Line 161 -> 206 will be export to an util function
+	keySecretRef := cr.Spec.ForProvider.ImportUrlSecretRef
+	importUrl := cr.Spec.ForProvider.ImportURL
+
+	// User has provided secret for the import
+	if keySecretRef != nil && importUrl != nil {
```

Comment:
Check also for actual non empty OR enforce minLength of Strings in CRD via kubebuilder annotation. Do both (non empty check AND minLength enforcement) for both parameters ideally.

Reply:
MinLength added for ImportUrl.
keySecretRef is not a string.